awscli/customizations/s3/utils.py

```diff
@@ -564,6 +565,7 @@ def map_copy_object_params(cls, request_params, cli_params):
         """Map CLI params to CopyObject request params"""
         cls._set_general_object_params(request_params, cli_params)
         cls._set_metadata_directive_param(request_params, cli_params)
+        cls._set_metadata_params(request_params, cli_params)
```

Comment:
We need to add it for multipart-uploads and copies as well. You can do this by setting it in ``map_create_multipart_upload``. It would be great to get an integration test as well to make sure multipart's are handled correctly.